homeassistant/components/group/binary_sensor.py

```diff
@@ -0,0 +1,133 @@
+"""This platform allows several binary sensor to be grouped into one binary sensor."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASSES_SCHEMA,
+    DOMAIN as BINARY_SENSOR_DOMAIN,
+    PLATFORM_SCHEMA,
+    BinarySensorEntity,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_DEVICE_CLASS,
+    CONF_ENTITIES,
+    CONF_NAME,
+    CONF_UNIQUE_ID,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import CoreState, Event, HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.typing import ConfigType
+
+from . import GroupEntity
+
+DEFAULT_NAME = "Binary Sensor Group"
+
+CONF_ALL = "all"
+REG_KEY = f"{BINARY_SENSOR_DOMAIN}_registry"
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITIES): cv.entities_domain(BINARY_SENSOR_DOMAIN),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+        vol.Optional(CONF_ALL): cv.boolean,
```

Comment:
We don't have an all option for the other group platforms. Eg see the light platform which also has a binary state. I think we should be really careful before we add new options to these platforms.

Reply:
I think these platforms should be refactored so the features are more similar and also to deduplicate common code.